ksqldb-functional-tests/src/test/resources/query-validation-tests/joins.json

```diff
@@ -3754,6 +3754,246 @@
         "type": "io.confluent.ksql.util.KsqlStatementException",
         "message": "Implicit repartitioning of windowed sources is not supported. See https://github.com/confluentinc/ksql/issues/4385."
       }
+    },
+    {
```

Comment:
The PR description makes it sound like this fix is for foreign key joins but none of these new test cases are for FK joins. Your fix is for all types of joins, right? Let's update the PR description (and title) to make that clear, and add a test case for an FK join as well (see `fk-joins.json`).

Reply:
Yes, the fix applies to all types of joins. I will update the description and add the the test in `fk-joins.json` as suggested.

Reply:
After investigating, foriegn-key joins with schema ids are not working with the fix in this PR. The error is also different as compared to other joins. I will open a new issue for fk-joins.

Reply:
Create issue - https://github.com/confluentinc/ksql/issues/9516